onnxruntime/contrib_ops/rocm/fused_conv.cc

```diff
@@ -81,41 +82,28 @@ struct FNVHash {
     // miopenConvolutionDescriptor_t, so we have to guess.
     // This algorithm is based on a specific behavior of miopenGetConvolutionNdDescriptor,
     //  which fails when requestedSpatialDim > the convolution's spatial dimension
-    std::vector<int> spatial_dims;
-    std::vector<int> pads;
-    std::vector<int> strides;
-    std::vector<int> dilations;
+    constexpr const int kMaxSpatialDim = 5;
+    std::vector<int> pads{kMaxSpatialDim};
+    std::vector<int> strides{kMaxSpatialDim};
+    std::vector<int> dilations{kMaxSpatialDim};
     miopenConvolutionMode_t mode;
-    while (true) {
-      spatial_dims.resize(spatial_dim);
-      pads.resize(spatial_dim);
-      strides.resize(spatial_dim);
-      dilations.resize(spatial_dim);
-
-      if (miopenStatusSuccess != miopenGetConvolutionNdDescriptor(cdesc,
-                                                                  spatial_dim,
-                                                                  spatial_dims.data(),
-                                                                  pads.data(),
-                                                                  strides.data(),
-                                                                  dilations.data(),
-                                                                  &mode)) {
-        // Remove the extra dimension
-        spatial_dims.resize(spatial_dim - 1);
-        pads.resize(spatial_dim - 1);
-        strides.resize(spatial_dim - 1);
-        dilations.resize(spatial_dim - 1);
-        (*this) << spatial_dim;
-        (*this) << spatial_dims;
-        (*this) << pads;
-        (*this) << strides;
-        (*this) << dilations;
-        break;
-      }
-      spatial_dim += 1;
-      ORT_ENFORCE(spatial_dim < 10,
-                  "miopenGetConvolutionNdDescriptor is supposed to fail before spatial_dim gets to ",
-                  spatial_dim);
+    bool spatial_dim_guessed = false;
+    for (int i=0; i<kMaxSpatialDim; i++) {
+      if (miopenStatusSuccess == miopenGetConvolutionNdDescriptor(
+        cdesc, i, &spatial_dim, pads.data(), strides.data(), dilations.data(), &mode)) {
```

Comment:
Why delete vector spatial_dims. I think the meaning of spatial_dim and spatial_dims are different.

Reply:
The original one was a miss use of API!